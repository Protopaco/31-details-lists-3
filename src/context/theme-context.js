import React, { createContext, useContext, useState, useEffect } from 'react'

export const ThemeContext = createContext(null);

export const ThemeProvider = ({ children }) => {
    const [theme, setTheme] = useState('light');

    const switchTheme = () => {
        if (theme === 'light') {
            setTheme('dark')
            document.documentElement.style.setProperty('--primary', 'brown');
            document.documentElement.style.setProperty('--secondary', 'beige');
        } else {
            setTheme('light')
            document.documentElement.style.setProperty('--primary', 'beige');
            document.documentElement.style.setProperty('--secondary', 'brown');
        }
    }

    return (
        <ThemeContext.Provider value={{ theme, switchTheme }}>
            {children}
        </ThemeContext.Provider>
    )
}

export const useTheme = () => {
    const { theme, switchTheme } = useContext(ThemeContext);
    return { theme, switchTheme }
}

