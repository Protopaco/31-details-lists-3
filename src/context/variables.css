:root{
    --primary: magenta;
    --secondary: rgb(1, 186, 174);
    --tertiary: tan;
}