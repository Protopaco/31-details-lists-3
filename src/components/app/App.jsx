import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import CountryByCode from '../../containers/CountryByCode';
import CountriesPage from '../../containers/CountriesPage';
import { usePagination } from '../../hooks/Countries';
import Header from '../../components/header/Header';
import { ThemeProvider } from '../../context/theme-context'



export default function App() {
  return (
    <Router>
      <ThemeProvider>
        <Header />
        <Switch>
          <Route exact path="/" component={CountriesPage} />
          <Route path="/country/:code" component={CountryByCode} />
        </Switch>
      </ThemeProvider>
    </Router>
  );
}
