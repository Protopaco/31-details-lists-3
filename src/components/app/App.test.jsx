import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { ThemeProvider } from '../../context/theme-context'

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(
      <ThemeProvider><App /></ThemeProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
