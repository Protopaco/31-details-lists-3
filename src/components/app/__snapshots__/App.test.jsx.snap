// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="backButton"
    >
      <a
        href="/"
      >
        <button
          class="homeButton"
        >
          HOME
        </button>
      </a>
      <button
        class="themeButton"
      >
        light
      </button>
    </div>
    <div
      class="text"
    >
      FLAGS OF THE WORLD
    </div>
  </div>
  <div
    class="frame"
  >
    <img
      alt="loading"
      src="http://static.demilked.com/wp-content/uploads/2016/06/gif-animations-replace-loading-screen-2.gif"
    />
  </div>
</DocumentFragment>
`;
